tests/Makefile

```diff
@@ -0,0 +1,283 @@
+SHELL = /bin/bash
+CC = gcc
+
+LIBS =
+PACKAGES =
+TEST_PROGRAMS =
+RESTRAINT_OBJS =
+
+PACKAGES += gio-2.0
+PACKAGES += gio-unix-2.0
+PACKAGES += glib-2.0
+PACKAGES += gobject-2.0
+PACKAGES += json-c
+PACKAGES += libarchive
+PACKAGES += libcurl
+PACKAGES += libsoup-2.4
+PACKAGES += libxml-2.0
+
+CFLAGS ?= -Og -g
+CFLAGS += -iquote ../src
+CFLAGS += -Wall -Werror -std=c99 $(shell pkg-config --cflags $(PACKAGES))
+
+ifeq ($(STATIC), 1)
+    # The right thing to do here is `pkg-config --libs --static`, which would
+    # include Libs.private in the link command.
+    # But really old pkg-config versions don't understand that so let's just
+    # hardcode the "private" libs here.
+    # The -( -) grouping means we don't have to worry about getting all the
+    # dependent libs in the right order (normally pkg-config would do that for
+    # us).
+    LIBS = -Wl,-Bstatic -Wl,-\( $(shell pkg-config --libs $(PACKAGES)) -lgmodule-2.0 -llzma -lbz2 -lz -lffi -lssl -lcrypto -lselinux -Wl,-\) -Wl,-Bdynamic -lm -pthread -lrt -lresolv -ldl -lutil $(LFLAGS)
+else
+    LIBS = $(shell pkg-config --libs $(PACKAGES) $(XTRAPKGS)) -lutil -pthread
+endif
+
+TEST_PROGRAMS += test_beaker_harness
+TEST_PROGRAMS += test_cmd_abort
+TEST_PROGRAMS += test_cmd_log
+TEST_PROGRAMS += test_cmd_result
+TEST_PROGRAMS += test_cmd_utils
+TEST_PROGRAMS += test_cmd_watchdog
+TEST_PROGRAMS += test_dependency
+TEST_PROGRAMS += test_env
+TEST_PROGRAMS += test_fetch_git
+TEST_PROGRAMS += test_fetch_uri
+TEST_PROGRAMS += test_logging
+TEST_PROGRAMS += test_metadata
+TEST_PROGRAMS += test_process
+#TEST_PROGRAMS += test_recipe
+#TEST_PROGRAMS += test_task
+TEST_PROGRAMS += test_utils
+
+.PHONY: all
+all: $(TEST_PROGRAMS)
+
+$(TEST_PROGRAMS): %: %.o
+	$(CC) $(LDFLAGS) -o $@ $? $(LIBS)
+
+### test_beaker_harness
+#
+BEAKER_HARNESS_OBJS =
+BEAKER_HARNESS_OBJS += beaker_harness.o
+
+RESTRAINT_OBJS += $(BEAKER_HARNESS_OBJS)
+
+MOCKS_BEAKER_HARNESS = -Wl,--wrap=soup_session_send_message
+
+test_beaker_harness: $(BEAKER_HARNESS_OBJS)
```

Comment:
Previously this also included test_beaker_harness.o and I'm not seeing that with this change.

Reply:
@cbouchar 
- This pull request is marked as draft
- As you may noticed it wasn't rebased for 25 days and it has multiple conflicts

Reply:
> @cbouchar
> 
> * This pull request is marked as draft
> * As you may notice it wasn't rebased for 25 days and it has multiple conflicts

Yes I noticed that which is why I quit reviewing it.

Reply:
> Previously the BEAKER_HARNESS_OBJS list included test_beaker_harness.o and I'm not seeing that with this change.

Previously there was no `BEAKER_HARNESS_OBJS` list, and as you can see in the rest of the file, these lists don't include test object files.

If the question is why the target `test_beaker_harness` doesn't explicitly include `test_beaker_harness.o`, the reason is because it's not needed, otherwise tests would fail to build.

That prerequisite is already included here,

```
$(TEST_PROGRAMS): %: %.o
	$(CC) $(LDFLAGS) -o $@ $^ $(LIBS)
```

The target you are commenting on is adding prerequisites from `BEAKER_HARNESS_OBJS`, as it's done in the other targets. And for this one, the recipe is overridden because of the "mocking".

> Yes I noticed that which is why I quit reviewing it.

Other hint was that no reviews were requested.